### ISSUE TITLE
error in process filter: open-network-stream: make client process failed: too many open files, 

### ISSUE THREAD

**Issue by vjohansen** [skeeto/elfeed#123]
From messages buffer

  error in process filter: open-network-stream: make client process failed: too many open files, :name,   papersincomputerscience.wordpress.com, :buffer, #<killed buffer>, :host, 
papersincomputerscience.wordpress.com, :service, 443, :nowait, nil
error in process filter: make client process failed: too many open files, :name, papersincomputerscience.wordpress.com, :buffer, #<killed buffer>, :host, papersincomputerscience.wordpress.com, :service, 443, :nowait, nil

I imported an opml file with a few hundred feeds and pressed G

GNU Emacs 24.5.1 (i686-pc-mingw32) of 2015-04-11 on LEG570
Windows 7

**Comment by vjohansen** [comment]
After restarting emacs and running elfeed , I get

    Debugger entered--Lisp error: (wrong-type-argument arrayp nil)
      elfeed-db-load()
      elfeed-db-ensure()
      #[0 "\304\305 \210\306\307\310\311\312\313\300\301\302#\314\"\315\316%\"\210\207" [(15552000 (unread) nil nil nil nil) ((nil)) (0) elfeed-db-index nil elfeed-db-ensure avl-tree-mapc make-byte-code 257 "\303!\304!\305\300\302\242$\205#

**Comment by skeeto** [comment]
Yeah, this is a big problem with the Windows version of Emacs. Since it
opens connections to so many different servers, it easily exceeds the
minuscule file limitations (presumably within MSVCRT). I'm not sure if
there's any way to fix this from the Elisp side of things. If you were
to successfully fetch without the first problem, you would also find
that Elfeed tends to crash Emacs a whole lot.

I think the followup problem occurred because Emacs, having opened too
many "files," was unable to properly write the database file and left a
corrupt database behind in its place.

It's probably not worth using Elfeed on Windows unless you keep to a
small number of feeds. Emacs isn't stable enough on that platform.

**Comment by vjohansen** [comment]
Would it be simple to implement that G only retrieves max N (oldest) feeds?

**Comment by skeeto** [comment]
<kbd>G</kbd> takes a prefix argument, only updating feeds for entries currently visible in the search buffer. You could use a filter to limit yourself to a certain subset of feeds, then <kbd>C-u G</kbd>.
